packages/permissionless/utils/getRequiredPrefund.ts

```diff
@@ -18,14 +20,36 @@ export type GetRequiredPrefundReturnType = {
  *     userOperation
  * })
  */
-export const getRequiredPrefund = ({
-    userOperation
-}: GetRequiredPrefundReturnType): bigint => {
-    const multiplier = userOperation.paymasterAndData.length > 2 ? 3n : 1n
+export const getRequiredPrefund = <entryPoint extends EntryPoint>({
+    userOperation,
+    entryPoint: entryPointAddress
+}: GetRequiredPrefundReturnType<entryPoint>): bigint => {
+    if (entryPointAddress === ENTRYPOINT_ADDRESS_V06) {
+        const userOperationVersion0_6 = userOperation as UserOperation<"0.6">
+        const multiplier =
+            userOperationVersion0_6.paymasterAndData.length > 2 ? 3n : 1n
+        const requiredGas =
+            userOperationVersion0_6.callGasLimit +
+            userOperationVersion0_6.verificationGasLimit * multiplier +
+            userOperationVersion0_6.preVerificationGas
+
+        return (
+            BigInt(requiredGas) * BigInt(userOperationVersion0_6.maxFeePerGas)
+        )
+    }
+
+    const userOperationVersion0_7 = userOperation as UserOperation<"0.7">
```

Comment:
-> userOperationVersionV07?

Reply:
V07 is confusing as it doesn't have point, we may have a a V1 in future too soo.